### PR TITLE
test(search): convert tests from cypress to playwright - FE-6152

### PR DESCRIPTION
**Pull request by DipperTheDan**
### Proposed behaviour

Tests should use Playwright

### Current behaviour

Tests use Cypress

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Additional context

N/A

### Testing instructions

N/A

### CODE REVIEW

**Review by edleeks87 — COMMENTED** [review]
Looks like we still have the [cypress locator files](https://github.com/Sage/carbon/tree/950b282155cc570565e2d0cfb59399262edd5c61/cypress/locators/search) to remove

File: src/components/search/search.pw.tsx [hunk 1]
```diff
@@ -0,0 +1,884 @@
...
+          </div>
+        </Box>
+      );
+
+      await checkAccessibility(page);
+    });
+  });
+
+  // FE-4670
+  test.skip("should render Search component", () => {
```
[the remark this answers is not in the document]
Comment (DipperTheDan) [reply src/components/search/search.pw.tsx:861]: Yeah it still fails accessibility checks. I'll use `Box` as we need to provide a background colour for this test.

**Review by edleeks87 — APPROVED** [review]

File: src/components/search/search.pw.tsx [hunk 2]
```diff
@@ -0,0 +1,866 @@
...
+const testDataStandard = CHARACTERS.STANDARD;
+const keysToTrigger = ["Enter", "Space"] as const;
+
+const validationTypes: [string, string][] = [
+  ["error", VALIDATION.ERROR],
+  ["warning", VALIDATION.WARNING],
+  ["info", VALIDATION.INFO],
+];
+
+test.describe("when focused", () => {
```
Comment (stephenogorman) [inline src/components/search/search.pw.tsx:35]:
```suggestion
test.describe("When focused", () => {

File: src/components/search/search.pw.tsx [hunk 3]
```diff
@@ -0,0 +1,866 @@
...
+    const searchCrossIconElementParent = searchCrossIcon(page).locator("..");
+
+    await expect(searchCrossIconElementParent).toHaveCSS(
+      "outline",
+      "rgb(255, 188, 25) solid 3px"
+    );
+  });
+});
+
+test.describe("check props for Search component", () => {
```
Comment (stephenogorman) [inline src/components/search/search.pw.tsx:168]:
```suggestion
test.describe("Prop tests for Search component", () => {

File: src/components/search/search.pw.tsx [hunk 4]
```diff
@@ -0,0 +1,866 @@
...
+
+    await expect(tooltipPreviewElement).toHaveText(testDataStandard);
+    await expect(tooltipPreviewElement).toHaveAttribute(
+      "data-placement",
+      String(tooltipPositionValue)
+    );
+  });
+});
+
+test.describe("check functionality for Search component", () => {
```
Comment (stephenogorman) [inline src/components/search/search.pw.tsx:536]:
```suggestion
test.describe("Functionality tests for Search component", () => {

File: src/components/search/search.pw.tsx [hunk 5]
```diff
@@ -0,0 +1,866 @@
...
+
+      const searchDefaultInputElement = searchDefaultInput(page);
+      await searchDefaultInputElement.press(key);
+
+      await expect(callbackCount).toEqual(1);
+    });
+  });
+});
+
+test.describe("accessibility tests for Search", () => {
```
Comment (stephenogorman) [inline src/components/search/search.pw.tsx:676]:
```suggestion
test.describe("Accessibility tests for Search", () => {

**Review by stephenogorman — CHANGES_REQUESTED** [review]

**Review by stephenogorman — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Pull request merged**